Architecture_Engine/Query/IsValidForDistanceCheck.cs

```diff
@@ -0,0 +1,100 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Architecture.BuildersWork;
+using BH.oM.Architecture.Elements;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.Engine.Geometry;
+using BH.oM.Geometry.CoordinateSystem;
+using System;
+
+namespace BH.Engine.Architecture
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static bool IsValidForDistanceCheck(this Opening opening1, Opening opening2, double maxDistance = double.NaN, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
+        {
+            // Check if the openings are parallel
+            if (opening1.CoordinateSystem.Z.IsParallel(opening2.CoordinateSystem.Z, angleTolerance) == 0)
+            {
+                return false;
+            }
+
+            // Check if the openigns are on the same plane/wall
+            Point centerPoint1 = opening1.CoordinateSystem.Origin;
+            Point centerPoint2 = opening2.CoordinateSystem.Origin;
+            Plane plane2 = new Plane { Origin = centerPoint2, Normal = opening2.CoordinateSystem.Z };
+
+            double distancePoint1Plane2 = centerPoint1.Distance(plane2);
+            double distanceDepths = opening1.Depth/2 + opening2.Depth/2;
+
+            if (distancePoint1Plane2 > distanceDepths + distanceTolerance)
+            {
+                return false;
+            }
+
+            // Check if the openings are closer than maxDistance
+            if (!double.IsNaN(maxDistance))
+            {
+                double diag1 = opening1.MaxDiagonal();
+                double diag2 = opening2.MaxDiagonal();
+
+                Point point1onPlane2 = centerPoint1.Project(plane2);
+
+                double c = centerPoint2.Distance(point1onPlane2);
+
+                if (maxDistance + (diag1/2 + diag2/2) + distanceTolerance < centerPoint2.Distance(point1onPlane2)) 
+                {
+                    return false;
+                }
+            }
+
+            return true;
+
+        }
+
```

Comment:
In BHoM we try to keep the code readable by separating the methods with decorators, on top of that each method type group (e.g. `private` vs `public`) should have its dedicated header - please feel free to explore other files for inspiration, in the meantime I post a suggestion how would I do it in this case.

```suggestion

        /***************************************************/
        /**** Private Methods                           ****/
        /***************************************************/
```